test/test.sql

```diff
@@ -0,0 +1,215 @@
+            AND BLL.EFFECTIVE_LOAD_DATE = DATE'2020-01-01'
+
+           LEFT JOIN SCHEMA.CRM_EXCHANGE_RATE E
+             ON E.EXCHANGE_RATE_DATE = DATE'2020-01-01'
+            AND E.EFFECTIVE_LOAD_DATE = DATE'2020-01-01'
+            AND E.TARGET_CURRENCY = BLL.CCY
+            AND E.EXCHANGE_RATE_CODE = 'FT0'
+
+           LEFT JOIN SCHEMA.CRM_PRODUCT P
+             ON P.PRODUCT = L.LOAN_CODE2
+            AND P.MODUL = 'CL'
+            AND NVL(P.RB_FORCED_LOAN, '#') = '#'
+            AND P.DML <> 'D'
+            AND P.START_OF_VALIDITY <= DATE'2020-01-01'
+            AND P.END_OF_VALIDITY > DATE'2020-01-01'
+
+          WHERE 1 = 1
+            AND AR.START_OF_VALIDITY <= DATE'2020-01-01'
+            AND AR.END_OF_VALIDITY > DATE'2020-01-01'
+            AND AR.ARRANGEMENT_TYPE = 'LOAN'
+          GROUP BY BLL.CUSTOMER_ID) SHORT_LONG
+ON SHORT_LONG.CUSTOMER_ID = C.CUSTOMER_ID
+LEFT JOIN (SELECT BLE.CUSTOMER_ID, SUM(BLE.FUTURE_CAPITAL) AS FI_CRED_LEASING_BAL, COUNT(LE.LEASING_ID) AS PR_CRED_LEASING_CNT
+
+           FROM SCHEMA.CRM_LEASING LE
+
+           JOIN SCHEMA.CRM_BALANCE_LEASING BLE
+             ON BLE.LEASING_ID = LE.LEASING_ID
+            AND BLE.CUSTOMER_ID IS NOT NULL
+            AND BLE.EFFECTIVE_LOAD_DATE = DATE'2020-01-01'
+
+          WHERE 1 = 1
+            AND LE.START_OF_VALIDITY <= DATE'2020-01-01'
+            AND LE.END_OF_VALIDITY > DATE'2020-01-01'
+
+          GROUP BY BLE.CUSTOMER_ID) DEALS_AND_INTERESTS
+ON DEALS_AND_INTERESTS.CUSTOMER_ID = C.CUSTOMER_ID
+LEFT JOIN (SELECT SCB.SYMBOLS_ID, COUNT(DISTINCT SCB.CONTRACT_ID) AS BPR_INS_LIFE_CNT, SUM(SCB.AMOUNT) AS FIN_INS_LIFE_BAL
+FROM SCHEMA.CRM_RL_SUBSIDIARY_COMPANIE_BAT SCB
+WHERE SCB.SOURCE_TCH = 'BIZTOSITO'
+AND SCB.EFFECTIVE_LOAD_DATE = &GLOBAL_P_EFFECTIVE_LOAD_DATE
+GROUP BY SCB.SYMBOLS_ID) LIFE_INSUR
+ON LIFE_INSUR.SYMBOLS_ID = C.SYMBOLS_CUSTOMER_SID
+
+LEFT JOIN (SELECT PD.CUSTOMER_ID, MAX(PD.DAYS_PAST_DUE) AS EXP_CRED_DAYS_MAX, ROUND(SUM(PD.PAST_DUE_AMOUNT)) AS EXP_CRED_AMT_MAX
+
+           FROM SCHEMA.CRM_PAST_DUE PD
+
+           JOIN SCHEMA.CRM_ARRANGEMENT AR
+             ON AR.ARRANGEMENT_ID = PD.ARRANGEMENT_ID
+            AND AR.START_OF_VALIDITY <= DATE'2020-01-01'
+            AND AR.END_OF_VALIDITY > DATE'2020-01-01'
+            AND AR.ARRANGEMENT_TYPE = 'LOAN'
+
+           JOIN SCHEMA.CRM_LOAN L
+             ON L.LOAN_ID = AR.LOAN_ID
+            AND L.START_OF_VALIDITY <= DATE'2020-01-01'
+            AND L.END_OF_VALIDITY > DATE'2020-01-01'
+            AND L.LOAN_CODE1 NOT LIKE 'T%'
+
+          WHERE 1 = 1
+            AND PD.PAST_DUE_SID4 = 'FACILITY_MAX_DPD'
+            AND PD.EFFECTIVE_LOAD_DATE = &GLOBAL_P_EFFECTIVE_LOAD_DATE
+          GROUP BY PD.CUSTOMER_ID) EXPIRED_LOAN
+ON EXPIRED_LOAN.CUSTOMER_ID = C.CUSTOMER_ID
+LEFT JOIN KMR_FINANCIAL_INDEX_STAGE_06 SZCH
+ON SZCH.CUSTOMER_ID = C.CUSTOMER_ID
+
+WHERE 1 = 1
+AND C.START_OF_VALIDITY <= DATE'2020-01-01'
+AND C.END_OF_VALIDITY > DATE'2020-01-01'
```

Comment:
Good luck refactoring this ~query~ Query 😁

Reply:
That's coming straight from #97, to be honest I didn't read it in depth, just compared the end result to expected output in tables aliases and tables 🙉

Reply:
That's the problem with examples coming from users - they tend to be not valid sql (wrong quotes, missing parenthesis etc), I don't know db2 dialect, but just checked and it should be date with () so will fix that

Reply:
Makes sense.

Reply:
Added () to date, there is still weird & (variable) but not sure how that should be handled